File: src/main/java/br/com/guizzo/projectbasic/modules/auth/domain/dto/AuthLoginDTO.java
package br.com.guizzo.projectbasic.modules.auth.domain.dto;


import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AuthLoginDTO{
        @NotNull(message = "E-mail inválido!")
        @Email(message = "Formato de e-mail incorreto!")
        private String email;

        @NotNull(message = "Senha inválida!")
        private String password;
}

File: src/modules/auth/domain/dto/auth-login.dto.ts
import { IsEmail } from 'class-validator';
import { IsPassword } from 'src/shared/decorators/is-password.decorator';

export class AuthLoginDto {
  @IsEmail({}, { message: 'E-mail inválido!' })
  email: string;

  @IsPassword()
  password: string;
}
